### DISCUSSION TITLE
Genarating user data export / cloning db table schema without data

### DISCUSSION THREAD

**Discussion by lumosminima** [pocketbase/pocketbase#5549] [Q&A] [answered]
I'm extending with go and have an endpoint: `/myapp/api/export/projectId="..."`. I want to implement user data export functionality and was wondering is there a way to get collections table raw db schema and use it to create tables in a new sqlite db. I want to create subset of db which only contains specific user's data (my app is multi tenant). I know I can manually create those tables but it could get out of sync and would add maintenance load. My plan is to use that table schema without _app_tables to create empty db using `dbx.Open` and then just use `app.Dao()` to populate the data between two using necessary filters and restrictions. This db file could then be imported and read in frontend using [sqlite-wasm+OPFS](https://sqlite.org/wasm/doc/trunk/persistence.md) and would allow user to view their past generated data or take it elsewhere.

I haven't done this before so am open to alternative way to achieve same goal.

**Comment by ganigeorgiev** [comment, marked as answer]
Please check https://pocketbase.io/docs/go-migrations/.

There is also a builtin command for generating migrations snapshots - https://pocketbase.io/docs/go-migrations/#collections-snapshot.
